### PR TITLE
Updated c++ and python shape examples after feedback.

### PR DESCRIPTION
**Pull request by cgalvan**
Made some improvements to the Shape Example tools after feedback:
- Added introduction text and link to the actual tutorial
- Modified placeholder text for entity name entry and added tooltip to suffix checkbox
- Added editable combo box example to set the scale on the entity

For just the python example, brought in some generic changes that were made to the base `PythonToolGem` template.

UpdatedShapeExamplesUX

Signed-off-by: Chris Galvan <chgalvan@amazon.com>

### CODE REVIEW

File: cpp_gems/ShapeExample/Code/Source/ShapeExampleWidget.cpp [hunk 1]
```diff
@@ -30,16 +34,31 @@ namespace ShapeExample
         setWindowTitle(QObject::tr("ShapeExample"));
 
         QVBoxLayout* mainLayout = new QVBoxLayout(this);
+        mainLayout->setSpacing(20);
 
-        QWidget* entityNameWidget = new QWidget(this);
+        // Introduction text explaining the example
+        QLabel* introText = new QLabel(QObject::tr("Welcome to the Python Shape Example tool. This tool demonstrates an example of creating an entity with a shape component in the editor. It also has other functional samples for you to play with too."), this);
```
Comment (AMZN-daimini) [inline cpp_gems/ShapeExample/Code/Source/ShapeExampleWidget.cpp:40]: nit: also-too in the same sentence :)

File: py_gems/PyShapeExample/Editor/Scripts/pyshapeexample_dialog.py [hunk 2]
```diff
@@ -41,23 +44,45 @@ def create_entity_with_shape_component(self, type_id):
...
         editor.EditorComponentAPIBus(bus.Broadcast, "AddComponentsOfType", new_entity_id, [type_id])
 
     def __init__(self, parent=None):
         super(PyShapeExampleDialog, self).__init__(parent)
 
         main_layout = QVBoxLayout()
+        main_layout.setSpacing(20)
+
+        # Introduction text explaining the example
+        intro_text = QLabel("Welcome to the Python Shape Example tool. This tool demonstrates an example of creating an entity with a shape component in the editor. It also has other functional samples for you to play with too.", self)
```
Comment (AMZN-daimini) [inline py_gems/PyShapeExample/Editor/Scripts/pyshapeexample_dialog.py:64]: nit: repeated from above (also-too).

**Review by AMZN-daimini — APPROVED** [review]

File: cpp_gems/ShapeExample/Code/Source/ShapeExampleWidget.cpp (hunk 1 above)
Comment (cgalvan) [reply cpp_gems/ShapeExample/Code/Source/ShapeExampleWidget.cpp:40]: Thanks will fix!

File: py_gems/PyShapeExample/Editor/Scripts/pyshapeexample_dialog.py (hunk 2 above)
Comment (cgalvan) [reply py_gems/PyShapeExample/Editor/Scripts/pyshapeexample_dialog.py:64]: Ditto!

**Pull request merged**